src/http/one/MessageParser.js

```diff
@@ -39,14 +41,70 @@ export default class MessageParser {
         }
 
         this.message = new this._messageType();
-        this.message.startLine.parse(match[1]);
-        this.message.header.parse(match[2]);
+        this.message.startLine = this.parseStartLine(match[1]);
+        this.message.header = this.parseHeader(match[2]);
         this.message.headerDelimiter = match[3];
         this._raw = match[4]; // body [prefix] or an empty string
 
         this.determineBodyLength();
     }
 
+    parseStartLine(/*raw*/) {
+        Must(false, "pure virtual: kids must override");
+    }
+
+    parseField(raw) {
+        let fieldRe = /^(.*?)([\t ]*:[\t ]*)(.*?)([\t \r]*\n)$/;
+        const match = fieldRe.exec(raw);
+
+        let field = new Field();
+
+        if (match) {
+            Must(match.length === 5);
+            field.name = match[1]; // right-trimmed
+            field.separator = match[2];
+            field.value = match[3]; // trimmed
+            field.terminator = match[4];
+        } else {
+            console.log(`Warning: Cannot parse ${raw.length}-byte header field: ${raw}`);
+            field.name = raw;
+            field.separator = "";
+            field.value = "";
+            field.terminator = "";
+        }
+
+        return field;
+    }
+
+    parseHeader(raw) {
+        let header = new Header();
+
+        Must(!header.fields.length);
+        Must(header._raw === null);
```

Comment:
Please remove the above two Must()s because we ought to trust the Header() constructor to do the right thing, whatever that is and cannot really know what the right thing is, exactly because we are not Header.